tests/snuba/search/test_backend.py

```diff
@@ -2051,7 +2059,209 @@ def test_query(query):
+                    "start_timestamp": iso_format(before_now(minutes=1)),
+                    "tags": {"my_tag": 1},
+                },
+                project_id=self.project.id,
+            )
+
+        transaction_event.group = self.group
+        self.perf_group_1 = Group.objects.get(id=transaction_event.group_id)
+        self.perf_group_1.update(type=GroupType.PERFORMANCE_SLOW_SPAN.value)
+
+        with mock.patch("sentry.event_manager._pull_out_data", hack_pull_out_data):
+            transaction_event = self.store_event(
+                data={
+                    **transaction_event_data,
+                    "event_id": "b" * 32,
+                    "timestamp": iso_format(before_now(minutes=1)),
+                    "start_timestamp": iso_format(before_now(minutes=1)),
+                    "tags": {"my_tag": 1},
+                },
+                project_id=self.project.id,
+            )
+
+        transaction_event.group = self.group
+        self.perf_group_2 = Group.objects.get(id=transaction_event.group_id)
+        self.perf_group_2.update(type=GroupType.PERFORMANCE_SLOW_SPAN.value)
+
+        error_event_data = {
+            "timestamp": iso_format(self.base_datetime - timedelta(days=20)),
+            "message": "bar",
+            "environment": "staging",
+            "tags": {
+                "server": "example.com",
+                "url": "http://example.com",
+                "sentry:user": "event2@example.com",
+                "my_tag": 1,
+            },
+        }
+
+        error_event = self.store_event(
+            data={
+                **error_event_data,
+                "fingerprint": ["put-me-in-error_group_1"],
+                "event_id": "c" * 32,
+                "stacktrace": {"frames": [{"module": "error_group_1"}]},
+            },
+            project_id=self.project.id,
+        )
+        self.error_group_1 = Group.objects.get(id=error_event.group.id)
+
+        error_event_2 = self.store_event(
+            data={
+                **error_event_data,
+                "fingerprint": ["put-me-in-error_group_2"],
+                "event_id": "d" * 32,
+                "stacktrace": {"frames": [{"module": "error_group_2"}]},
+            },
+            project_id=self.project.id,
+        )
+        self.error_group_2 = Group.objects.get(id=error_event_2.group.id)
+
+    def test_performance_query(self):
+        with self.feature("organizations:performance-issues"):
+            results = self.make_query(search_filter_query="issue.category:performance my_tag:1")
+        assert set(results) == {self.perf_group_1}
+
+    def test_error_performance_query(self):
+        with self.feature("organizations:performance-issues"):
+            results = self.make_query(search_filter_query="my_tag:1")
+        assert set(results) == {
+            self.perf_group_1,
+            self.error_group_1,
+            self.perf_group_2,
+            self.error_group_2,
+        }
```

Comment:
`perf_group_1` and `perf_group_2` are the same group here, so the set just de-dupes them. Probably safer to use a list to compare them.